assets/app/scripts/services/applicationGenerator.js

```diff
@@ -86,17 +73,20 @@ angular.module("openshiftConsole")
         imageStream: scope._generateImageStream(input),
         buildConfig: scope._generateBuildConfig(input, imageSpec, input.labels),
         deploymentConfig: scope._generateDeploymentConfig(input, imageSpec, ports, input.labels),
-        service: scope._generateService(input, input.name, scope._getFirstPort(ports))
+        service: scope._generateService(input, input.name, ports)
       };
       resources.route = scope._generateRoute(input, input.name, resources.service.metadata.name);
       return resources;
     };
 
     scope._generateRoute = function(input, name, serviceName){
+      var route;
+
       if(!input.routing.include) {
         return null;
       }
-      return {
+
+      route = {
```

Comment:
Been declaring vars at the top a la Douglas Crockford, but I don't have a strong opinion on it. I'll declare it inline.

Should we have a style guide for our JS?